vmx/src/memory_path.rs

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use crate::error::Error::{self, PersistenceError};
+
+#[derive(Debug)]
+pub struct MemoryPath {
+    path: PathBuf,
+}
+
+impl MemoryPath {
+    pub fn new<P: AsRef<Path>>(path: P) -> Self
+    where
+        P: Into<PathBuf>,
+    {
+        MemoryPath { path: path.into() }
+    }
+
+    pub fn read(&self) -> Result<Vec<u8>, Error> {
```

Comment:
I don't see the reason for reading the whole file into memory when we already have the memory maps.

Reply:
done